### PR TITLE
Improved map panel and map card to ignore zones when fitting map.

### PR DESCRIPTION
**Pull request by klejejs**
Fixed this issue  [#1598](https://github.com/home-assistant/home-assistant-polymer/issues/1598).

### CODE REVIEW

File: src/panels/map/ha-panel-map.js [hunk 1]
```diff
@@ -69,8 +69,13 @@ class HaPanelMap extends LocalizeMixin(PolymerElement) {
 
   fitMap() {
     var bounds;
+    var devices = [];
 
-    if (this._mapItems.length === 0) {
+    this._mapItems.forEach((item) => {
+      if (!item.options.zone) devices.push(item.getLatLng());
+    });
```
Comment (bramkragten) [inline src/panels/map/ha-panel-map.js]:
```js
const devices = this._mapItems.filter((item) => !item.options.zone);
```

File: src/panels/map/ha-panel-map.js [hunk 2]
```diff
@@ -69,8 +69,13 @@ class HaPanelMap extends LocalizeMixin(PolymerElement) {
 
   fitMap() {
     var bounds;
+    var devices = [];
 
-    if (this._mapItems.length === 0) {
+    this._mapItems.forEach((item) => {
+      if (!item.options.zone) devices.push(item.getLatLng());
```
Comment (bramkragten) [inline src/panels/map/ha-panel-map.js]:
```suggestion
      if (!item.options.zone) {
          devices.push(item.getLatLng());
      }
```

File: src/panels/lovelace/cards/hui-map-card.ts [hunk 3]
```diff
@@ -248,7 +248,16 @@ class HuiMapCard extends LitElement implements LovelaceCard {
       return;
     }
     const zoom = this._config.default_zoom;
-    if (this._mapItems.length === 0) {
+
+    const devices = new Array();
+    this._mapItems.forEach((item) => {
+      // @ts-ignore
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-map-card.ts]: We shouldn't ignore it, but extend the interface.

**Comment by bramkragten** [comment]
Instead of adding it as an option to the marker, we could also save them in a different array when creating them and use that for the fitting.

**Comment by klejejs** [comment]
Ok, so I took your idea, reverted to the previous version and created a new array for zones, which is not used for fitting map.

**Review by bramkragten — APPROVED** [review]

**Comment by klejejs** [comment]
Can you please close the issue? (#1598 )

**Pull request merged**